### ISSUE TITLE
为啥损失函数是CE交叉熵损失函数，就负样本为false？？加了以后模型效果还变差了。

### ISSUE THREAD

**Issue by MRZHAOWC** [RUCAIBox/RecBole#1133]
**Describe the bug**

**To Reproduce**
1. extra yaml file
2. your code
3. script for running

**Expected behavior**

**Screenshots**

**Colab Links**
If applicable, add links to Colab or other Jupyter laboratory platforms that can reproduce the bug.

**Desktop (please complete the following information):**
 - OS: [e.g. Linux, macOS or Windows]
- RecBole Version [e.g. 0.1.0]
 - Python Version [e.g. 3.79]
- PyTorch Version [e.g. 1.60]
- cudatoolkit Version [e.g. 9.2, none]

**Comment by Sherry-XLL** [comment]
@MRZHAOWC Generally, BPR loss function or CE loss function needs to be set in the sequential recommendation model. `loss_type (str)` denotes the type of loss function. If `loss_type='CE'`, the training task is regarded as a multi-classification task and the target item is the ground truth. In this way, negative sampling is not needed. If `loss_type='BPR'`, the training task will be optimized in the pair-wise way, which maximize the difference between positive item and negative item. In this way, negative sampling is necessary, such as setting `--neg_sampling="{'uniform': 1}"`. Defaults to `'CE'`. Range in `['BPR', 'CE']`.

The loss function determines the optimization method of the function, but there is no absolute advantage or disadvantage. Under different data sets, models and hyper parameter settings, different loss functions will have different performance results.

What's more, if you want to use negative sampling in the cross entropy loss, you can use `BCE` as the loss function in the pair-wise way as follows.

```python
# class torch.nn.BCELoss(weight: Optional[torch.Tensor] = None, size_average=None, reduce=None, reduction: str = 'mean')
elif self.loss_type == 'BCE':
    neg_items = interaction[self.NEG_ITEM_ID]
    pos_items_emb = self.item_embedding(pos_items)
    neg_items_emb = self.item_embedding(neg_items)
    pos_score = torch.sum(seq_output * pos_items_emb, dim=-1)  # [B]
    neg_score = torch.sum(seq_output * neg_items_emb, dim=-1)  # [B]
    predict = torch.cat((pos_score, neg_score))
    target = torch.zeros(len(pos_items) + len(neg_items), dtype=torch.float32).to(self.device)
    target[:len(pos_items)] = 1
    loss = nn.BCELoss(predict, target)
    return loss
```

**Comment by MRZHAOWC** [comment]
好的，谢谢回答。我明白了。
